pkg/controllers/jobset_controller.go

```diff
@@ -701,6 +704,10 @@ func GenSubdomain(js *jobset.JobSet) string {
 	return js.Name
 }
 
+func namespacedJobName(ns string, jobName string) string {
+	return fmt.Sprintf("%s-%s", ns, jobName)
```

Comment:
```suggestion
	return fmt.Sprintf("%s/%s", ns, jobName)
```

Reply:
I used a `/` originally but there was a bunch of errors indicating that this character is not allowed in labels, so I switched to `-`.

Reply:
sg

Reply:
I am concerned about the length of the key, perhaps we should hash it to limit its length and call it job-key instead of job-name?

Reply:
done

Reply:
I used sha1 hash since the hex digest is 40 chars, less than our 63 char limit.